### PR TITLE
Replace cross-ns owner refs with finalizer.

### PR DESCRIPTION
**Pull request by markusthoemmes**
This fixes SRVKS-344. Cross-namespace owner references are a nono.

### CODE REVIEW

**Comment by savitaashture** [comment]
Thank you :+1: 

/lgtm

File: pkg/controller/knativeserving/knativeserving_controller.go [hunk 1]
```diff
@@ -134,9 +134,6 @@ func (r *ReconcileKnativeServing) Reconcile(request reconcile.Request) (reconcil
 	instance := &servingv1alpha1.KnativeServing{}
 	if err := r.client.Get(context.TODO(), request.NamespacedName, instance); err != nil {
```
Comment (jcrossley3) [inline pkg/controller/knativeserving/knativeserving_controller.go:135]: Why do we need this condition at all anymore?

File: pkg/controller/knativeserving/knativeserving_controller.go [hunk 2]
```diff
@@ -219,6 +232,21 @@ func (r *ReconcileKnativeServing) install(instance *servingv1alpha1.KnativeServi
 	return nil
 }
 
+func (r *ReconcileKnativeServing) delete(instance *servingv1alpha1.KnativeServing) error {
```
Comment (jcrossley3) [inline pkg/controller/knativeserving/knativeserving_controller.go:235]: Don't we still need the `isInteresting` query in here somewhere?

File: pkg/controller/knativeserving/knativeserving_controller.go (hunk 1 above)
Comment (markusthoemmes) [reply pkg/controller/knativeserving/knativeserving_controller.go:135]: Well yes, we load the instance here so we definitely need this call. On notfound we just want to abort, any other errors we want to surface.

File: pkg/controller/knativeserving/knativeserving_controller.go (hunk 2 above)
Comment (markusthoemmes) [reply pkg/controller/knativeserving/knativeserving_controller.go:235]: Nope, we only enter here after we already checked it above.

Comment (jcrossley3) [reply pkg/controller/knativeserving/knativeserving_controller.go:235]: pb

**Comment by jcrossley3** [comment]
I like it -- much cleaner

**Pull request merged**